src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int AMOUNT_OF_STORAGE = 10;
+    private static final int NOT_FIND = -1;
+    private Object[] objKey;
+    private Object[] objValue;
+    private int realLengthStorage;
+
+    public StorageImpl() {
+        objKey = new Object[AMOUNT_OF_STORAGE];
+        objValue = new Object[AMOUNT_OF_STORAGE];
+        realLengthStorage = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+
+        if (get(key) == null) {
+            objKey[realLengthStorage] = key;
+            objValue[realLengthStorage] = value;
+            realLengthStorage++;
+        } else {
+            objValue[realLengthStorage - 1] = value;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int searchKey = NOT_FIND;
+        for (int i = 0; i < realLengthStorage; i++) {
+            if ((key == null && objKey[i] == null)
+                    || ((objKey[i] != null) && (objKey[i].equals(key)))) {
```

Comment:
Let's remove redundant brackets
```suggestion
            if ((key == null && objKey[i] == null)
                    || ((objKey[i] != null) && (objKey[i].equals(key)))) {
```
And let's think about how can we shorten this expression

Reply:
Everything I could to do make shorter is: optimization method and pass check boolean expression to other method.